peer/brontide.go

```diff
@@ -4153,6 +4160,46 @@ func (p *Brontide) handleRemovePendingChannel(req *newChannelMsg) {
 	p.addedChannels.Delete(chanID)
 }
 
+// handlePeerStorageMessage handles `PeerStorage` message, it stores the message
+// and sends it back to the peer as an ack.
+func (p *Brontide) handlePeerStorageMessage(msg *lnwire.PeerStorage) error {
```

Comment:
suggestion for function structure: I like to use the "exit early"/"golden path" rule as I think it makes things easier to reason since it avoids needing to have many nested `if` blocks. So I suggest:

```
func (p *Brontide) handlePeerStorageMessage(msg *lnwire.PeerStorage) error {
	if p.cfg.PeerDataStore == nil {
		// Send back warning if we are currently not handling
		// peerStorage message but received peerStorage message.
		warning := &lnwire.Warning{
			ChanID: lnwire.ConnectionWideID,
			Data: []byte("received peer storage message but not " +
				"advertising required feature bit"),
		}

		// Disconnect?

		return p.SendMessage(false, warning)
	}

	if !p.hasActiveChannels() {
		p.log.Tracef("Received peerStorage message from "+
			"peer(%v) with no active channels -- ignoring",
			p.String())

		return nil
	}

	// Store the data.
	err := p.cfg.PeerDataStore.Store(msg.Blob)
	if err != nil {
		return err
	}

	p.log.Debugf("Persisting peerbackup for peer(%s)", p)

	return nil
}
```